src/modals/SettingsModal/GeneralSettingsSection.tsx

```diff
@@ -16,21 +16,40 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
+import { useCallback, useState } from 'react'
+
 import KeyValueInput from '../../components/Inputs/InlineLabelValueInput'
 import Input from '../../components/Inputs/Input'
 import Toggle from '../../components/Inputs/Toggle'
 import HorizontalDivider from '../../components/PageComponents/HorizontalDivider'
+import PasswordConfirmation from '../../components/PasswordConfirmation'
 import ThemeSwitcher from '../../components/ThemeSwitcher'
 import { useGlobalContext } from '../../contexts/global'
+import CenteredModal from '../CenteredModal'
 
 const GeneralSettingsSection = () => {
   const {
     settings: {
       general: { walletLockTimeInMinutes, discreetMode, passwordRequirement }
     },
-    updateSettings
+    updateSettings,
+    wallet
   } = useGlobalContext()
 
+  const [requiresPasswordConfirmation, setRequiresPasswordConfirmation] = useState(false)
```

Comment:
I would recommend renaming this so that it is consistent with our [other modals](https://github.com/alephium/desktop-wallet/blob/master/src/pages/Wallet/WalletLayout.tsx#L55). I'd also [wrap the modal with the framer component](https://github.com/alephium/desktop-wallet/blob/master/src/pages/Wallet/WalletLayout.tsx#L138-L150) to make closing it more smooth:

```diff
diff --git a/src/modals/SettingsModal/GeneralSettingsSection.tsx b/src/modals/SettingsModal/GeneralSettingsSection.tsx
index 32cc621..29ade4a 100644
--- a/src/modals/SettingsModal/GeneralSettingsSection.tsx
+++ b/src/modals/SettingsModal/GeneralSettingsSection.tsx
@@ -16,6 +16,7 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
+import { AnimatePresence } from 'framer-motion'
 import { useCallback, useState } from 'react'
 
 import KeyValueInput from '../../components/Inputs/InlineLabelValueInput'
@@ -36,10 +37,11 @@ const GeneralSettingsSection = () => {
     wallet
   } = useGlobalContext()
 
-  const [requiresPasswordConfirmation, setRequiresPasswordConfirmation] = useState(false)
+  const [isPasswordModelOpen, setIsPasswordModalOpen] = useState(false)
+
   const onPasswordRequirementChange = useCallback(() => {
     if (passwordRequirement) {
-      setRequiresPasswordConfirmation(true)
+      setIsPasswordModalOpen(true)
     } else {
       updateSettings('general', { passwordRequirement: true })
     }
@@ -47,7 +49,7 @@ const GeneralSettingsSection = () => {
 
   const disablePasswordRequirement = useCallback(() => {
     updateSettings('general', { passwordRequirement: false })
-    setRequiresPasswordConfirmation(false)
+    setIsPasswordModalOpen(false)
   }, [updateSettings])
 
   return (
@@ -93,15 +95,17 @@ const GeneralSettingsSection = () => {
           <HorizontalDivider narrow />
         </>
       )}
-      {requiresPasswordConfirmation && (
-        <CenteredModal title="Password" onClose={() => setRequiresPasswordConfirmation(false)} focusMode>
-          <PasswordConfirmation
-            text="Type your password to change this setting."
-            buttonText="Enter"
-            onCorrectPasswordEntered={disablePasswordRequirement}
-          />
-        </CenteredModal>
-      )}
+      <AnimatePresence>
+        {isPasswordModelOpen && (
+          <CenteredModal title="Password" onClose={() => setIsPasswordModalOpen(false)} focusMode>
+            <PasswordConfirmation
+              text="Type your password to change this setting."
+              buttonText="Enter"
+              onCorrectPasswordEntered={disablePasswordRequirement}
+            />
+          </CenteredModal>
+        )}
+      </AnimatePresence>
     </>
   )
 }
```